src/polyhedralGravity/model/Polyhedron.cpp

```diff
@@ -205,18 +210,11 @@ namespace polyhedralGravity {
         const Array3 rayOrigin = centroid + (rayVector * EPSILON_ZERO_OFFSET);
 
         // Count every triangular face which is intersected by the ray
-        const auto &[begin, end] = this->transformIterator();
-        std::set<Array3> intersections{};
-        std::for_each(begin, end, [&rayOrigin, &rayVector, &intersections](const Array3Triplet &otherFace) {
-            const std::unique_ptr<Array3> intersection = rayIntersectsTriangle(rayOrigin, rayVector, otherFace);
-            if (intersection != nullptr) {
-                intersections.insert(*intersection);
-            }
-        });
-        return intersections.size();
+        return this->_tree->countIntersections(rayOrigin, rayVector);
     }
 
-    std::unique_ptr<Array3> Polyhedron::rayIntersectsTriangle(const Array3 &rayOrigin, const Array3 &rayVector, const Array3Triplet &triangle) {
+    //TODO: moved to LeafNode.cpp, consider removing here
```

Comment:
I want to remove the intersection logic from the Polyhedron and move it to the LeafNode since the tree will be responsible for intersection calculation in the future. Since Polyhedron is in the forked repo I wanted to communicate this change before putting it into action.

Reply:
I'm okay with any change; that's why you are working on a fork/ branch.

------

Also, please keep an eye on the size of the Pull Request. It's quite difficult to review a PR with more than 1000 lines of code being changed, especially if there are multiple review rounds.

Of course, an initial "green-field-engineering" solution will always be a big chunk at once (so no issue, I also did this with my supervisors back then in my thesis 😜)

However, keep the reasonable-sized PR thing in mind and consider starting a new branch at some point (and this getting merged into the fork's `main`)
A reasonable size means something like 200 - 400 lines of code have been changed.

Reply:
Thanks, I will do that.